### ISSUE TITLE
Tileset with individual image + collision, issue with index computation

### ISSUE THREAD

**Issue by mel-mouk** [RafaelBarbosatec/bonfire#417]
Hi,

I'm currently working with a dynamically built tileset. To simplify my tileset edition, I use individual images & collision definitions. My tiles will be something like this: 
```
{
        "id":4,
	"image":"some-tile.png",
	"imageheight":32,
	"imagewidth":32,
	"objectgroup": {
	        "type":"objectgroup",
		"objects":[
			{
				"id":1,
				"height":32,
				"width":16,
				"rotation":0,
				"x":16,
				"y":0
			}
		],
	}
```

My collisions were not considered, and after further investigation, I noticed something weird in `tiled_world_builder.dart`. 

In `_getDataTile` line 328, you override firsTgId value if the tileset contains an image to compute the image position properly, but doing that breaks the computation for collisions because you try to look for a collision using index - firsTgId after firsTgId was updated. I didn't try it yet but I think it also breaks animation.

Locally, I made a working fix by storing an `originalFirsTgId` value that I use to call _getCollision instead of firsTgId. 

Is it ok for you if I create a PR with that change?

**Comment by RafaelBarbosatec** [comment]
Hi @mel-mouk ! 
Thanks for this report!
Please, would be awesome you open the PR for this solution.
😊

**Comment by RafaelBarbosatec** [comment]
Available in version 1.12.7
